skipchain/skipchain_test.go

```diff
@@ -616,3 +579,44 @@ func makeHELS(local *onet.LocalTest, nbr int) ([]*onet.Server, *onet.Roster, *Se
 	el := local.GenRosterFromHost(hosts...)
 	return hosts, el, local.Services[hosts[0].ServerIdentity.ID][skipchainSID].(*Service)
 }
+
+func waitPropagationFinished(local *onet.LocalTest) {
+	var servers []*onet.Server
+	for _, s := range local.Servers {
+		servers = append(servers, s)
+	}
+	services := make([]*Service, len(servers))
+	for i, s := range local.GetServices(servers, skipchainSID) {
+		services[i] = s.(*Service)
+	}
+	propagating := true
+	for propagating {
+		propagating = false
+		for _, s := range services {
+			if s.IsPropagating() {
+				log.Lvl1("Service", s, "is still propagating")
+				propagating = true
+			}
+		}
+		if propagating {
+			time.Sleep(time.Millisecond * 100)
+		}
+	}
+}
+
+func checkBacklinks(services []*Service, sb *SkipBlock) {
```

Comment:
what's the reason for this here ? I mean, this check is already supposed to happen somewhere else right ?

Reply:
It used to be in two tests, no only one of them is remaining. But I wanted to clean up a bit and put all private functions to the bottom of the file. I'll inline it, as it's only used once now.